sklearn/metrics/tests/test_score_objects.py

```diff
@@ -973,3 +973,29 @@ def test_scorer_no_op_multiclass_select_proba():
         roc_auc_score, needs_proba=True, multi_class="ovo", labels=lr.classes_,
     )
     scorer(lr, X_test, y_test)
+
+
+def test_scorer_general_multiclass():
+    X, y = make_classification(
+        n_classes=3, n_informative=3, n_samples=30, random_state=0
+    )
+
+    # use `stratify` = y to ensure train and test sets capture all classes
+    X_train, X_test, y_train, y_test = train_test_split(
+        X, y, random_state=0, stratify=y
+    )
+
+    tree = DecisionTreeClassifier(random_state=0).fit(X_train, y_train)
+    lr = LogisticRegression(random_state=0).fit(X_train, y_train)
+
+    # don't use these scorers since they're not for multiclass
+    binary_only = [
+        'top_k_accuracy', 'f1', 'roc_auc', 'average_precision', 'precision',
+        'recall', 'neg_log_loss', 'neg_brier_score', 'jaccard'
+    ]
+
+    for name in CLF_SCORERS:
+        if name not in binary_only:
+            scorer = SCORERS[name]
+            scorer(tree, X_test, y_test)
```

Comment:
Instead of a smoke test, we should probably do a check that the scorer give the same than the calling the score function. For instance, in the binary case, we do:    

```python
score1 = get_scorer('%s' % prefix)(clf, X_test, y_test)
score2 = metric(y_test, clf.predict(X_test), pos_label=1)
assert_almost_equal(score1, score2)
```